src/main/java/com/internship/socialnetwork/repository/FriendshipRepository.java

```diff
@@ -0,0 +1,36 @@
+package com.internship.socialnetwork.repository;
+
+import com.internship.socialnetwork.model.Friendship;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface FriendshipRepository extends JpaRepository<Friendship, Long> {
+
+    @Query("SELECT friendship " +
+            "FROM Friendship friendship " +
+            "WHERE (friendship.user.id = :userId AND friendship.friendshipStatus = ACCEPTED) OR " +
+            "(friendship.friend.id = :userId AND friendship.friendshipStatus = ACCEPTED) ")
+    List<Friendship> findAllFriends(Long userId);
+
+    @Query("SELECT friendship " +
+            "FROM Friendship friendship " +
+            "WHERE friendship.friend.id = :userId AND friendship.friendshipStatus = PENDING")
+    List<Friendship> findAllFriendRequests(Long userId);
```

Comment:
You can merge this query with the one below it by accepting status parameter

Reply:
Are you sure? First one has friendship.friend and second one friendship.user. Status is same in both of them.

Reply:
You're right, didn't see that